### PR TITLE
feat: add SubCommand `copliot` to complete the auto-battle feature 

### PR DESCRIPTION
**Pull request by hzxjy1**
In fact, it is PR #103. But I had done something wrong that had forced the closure of PR #103 when I tried to rebase the newest commit. 

In the new version, it uses `BoolInput` to receive the user's selection. Moreover, it adds two features. One is the ability to identify JSON files, and the other is the capability to read the clipboard with the option "--paste".

In passing, should "Cargo.lock" be tracked by git? Like "package-lock.json" of node.

### CODE REVIEW

**Comment by wangl-cc** [comment]
The compile failure appears to be caused by the clipboard dependency, possibly due to a lack of some system dependencies. I'm uncertain about the usefulness of reading from the paste, and introducing this dependency might not be warranted.

**Comment by hzxjy1** [comment]
If it causes too much trouble, removing it is also a good option.

File: maa-cli/Cargo.toml [hunk 1]
```diff
@@ -42,10 +42,10 @@ futures-util = { version = "0.3.28" }
 sha2 = { version = "0.10.7" }
 digest = { version = "0.10.7" }
 semver = { version = "1.0.19", features = ["serde"] }
-
 # Dependencies used to extract files
 flate2 = { version = "1", optional = true }
 tar = { version = "0.4.40", optional = true }
+prettytable = "0.10.0"
```
Comment (wangl-cc) [inline maa-cli/Cargo.toml:48]: The `prettytable` crate depends on `csv` by default, providing functions for importing and exporting CSV files, which we don't use. I suggest we use the non-default feature. Additionally, we can consider moving this dependency higher in the list, as this group includes some optional dependencies.

**Comment by wangl-cc** [comment]
> In passing, should "Cargo.lock" be tracked by git? Like "package-lock.json" of node.

Regarding `Cargo.lock`, for a binary, it should be included because the binary needs to be compiled with `cargo build --locked ....`

**Comment by hzxjy1** [comment]
Done👌.

**Comment by wangl-cc** [comment]
I've just tested this, and it looks good to me.

**Comment by wangl-cc** [comment]
Thanks for your contribution.

**Comment by hzxjy1** [comment]
ohhhh

**Comment by Cryolitia** [comment]
`cargo.lock` is vital, indispensable for Nix build system.

**Pull request merged**